pywbem_mock/_mainprovider.py

```diff
@@ -811,152 +790,247 @@ def remove_namespace(self, namespace):
     #
     ####################################################################
 
-    def EnumerateClasses(self, **params):
-        # pylint: disable=invalid-name
+    def EnumerateClasses(self, namespace, ClassName=None,
+                         DeepInheritance=None, LocalOnly=None,
+                         IncludeQualifiers=None, IncludeClassOrigin=None):
         """
-        Implements a mock server responder for
+        Implements a CIM server responder for
         :meth:`~pywbem.WBEMConnection.EnumerateClasses`.
 
-        Enumerate classes from class repository. If classname parameter
+        Enumerate classes from datastore. If classname parameter
         exists, use it as the starting point for the hiearchy to get subclasses.
+        Parameters:
```

Comment:
Here you could refer to the parameters of `FakeWBEMConnection.EnumerateClasses` instead of repeating them. In addition, it would be good to have a sentence about using the parameter defaults defined in DSP0200.

Reply:
There is no definition of FakedWBEMConnection.EnumerateClasses.  The server method is defined here and because of the differences we introduced between the client and server methods (namespace arg is positional, ClassNams is string only, etc. We really do need to precisely define the methods in MainProvider and DefaultInstanceProvider.

Reply:
ok

Reply:
DONE or DISCUSSION